src/python_testing/TC_BOOLCFG_3_1.py

```diff
@@ -65,40 +66,41 @@ async def test_TC_BOOLCFG_3_1(self):
 
         self.step("2a")
         feature_map = await self.read_boolcfg_attribute_expect_success(endpoint=endpoint, attribute=attributes.FeatureMap)
+        is_sens_level_feature_supported = feature_map & Clusters.BooleanStateConfiguration.Bitmaps.Feature.kSensitivityLevel
 
         self.step("2b")
-        attribute_list = await self.read_boolcfg_attribute_expect_success(endpoint=endpoint, attribute=attributes.AttributeList)
+        if not is_sens_level_feature_supported:
+            logging.info("SENS feature not supported, skipping test case")
 
-        is_sens_level_feature_supported = feature_map & Clusters.BooleanStateConfiguration.Bitmaps.Feature.kSensitivityLevel
+            # Skipping all remainig steps
+            for step in self.get_test_steps(self.current_test_info.name)[self.current_step_index:]:
+                self.step(step.test_plan_number)
```

Comment:
There's a skip_remaining_steps command now in the base that will mark these properly in the TH. And it's a one liner.

Reply:
Oh nice. Is it ok with you if I do that as a follow up improvement PR for all the test cases containing the current loop?